### PR TITLE
Pcc 90 x

### PR DESCRIPTION
**Pull request by nfilipov**
This PR affects only the Pixel Cluster Counting method. In this PR we:
- add a switch to be respectful of the Phase2 tracker geometry (number of pixel barrel layers)
- remove some Run 1 unused stuff (commented out already)
- replace the TrackerGeometry::DetContainer loop, with an edmNew::DetSet<SiPixelCluster> loop that has the same effect.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoLuminosity/LumiProducer/test/analysis/plugins/PCCNTupler.cc [hunk 1]
```diff
@@ -315,65 +310,61 @@ void PCCNTupler::analyze(const edm::Event& iEvent,
...
     }
 
-
-    // -- Does this belong into beginJob()?
-    ESHandle<TrackerGeometry> TG;
-    iSetup.get<TrackerDigiGeometryRecord>().get(TG);
-
+    int NumPixelBarrelLayers=3;
+    if(pixelPhase2Geometry){
+      NumPixelBarrelLayers=4;
```
Comment (slava77) [inline RecoLuminosity/LumiProducer/test/analysis/plugins/PCCNTupler.cc]: I'm curious why is this only for phase2?
Doesn't phase-1 have the same feature?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17647	3cb09eb
- test/analysis code, not used in standard/central workflows
- jenkins tests pass: compiles

**Comment by davidlange6** [comment]
+1

**Pull request merged**